src/Mod/Assembly/App/AssemblyObject.cpp

```diff
@@ -1072,6 +1076,13 @@ std::string AssemblyObject::handleOneSideOfJoint(App::DocumentObject* joint,
     App::DocumentObject* part = getLinkObjFromProp(joint, propPartName);
     App::DocumentObject* obj = getObjFromNameProp(joint, propObjName, propPartName);
 
+    if (!part) {
+        std::string msg = std::string("Assembly: The property ") + propPartName + " of Joint "
+            + joint->getFullName() + " is empty.\n";
+        Base::Console().Error(msg.c_str());
```

Comment:
For security reasons never do this but use a format string: `Base::Console().Error("%s\n", msg.c_str());`

Reply:
Thanks,. I didn't find Assembly workbench domain specific exception classes, so I use ValueError in the update (see below). Also, the message is now untranslated - it contain parameters and I am not sure if there is support for translation of parametrized messages in FreeCAD. It woudn't be bad to have it also translated.

Reply:
Do we want to throw exception here? `handleOneSideOfJoint` is called from `makeMbdJoint` and it already returns `{}` if joint cannot be created. Anyway, it seems to solve only consequences. @PaddleStroke?

Reply:
This issue is actually fixed in my current dev branch. Incomplete joints are automatically removed. So this is not necessary. Though an additional check cannot hurt I guess. Though instead of throwing an error, printing a warning and ignoring the joint is better.